app/src/main/java/br/digitalhouse/marveltime/CadastroActivity.java

```diff
@@ -0,0 +1,171 @@
+        else
+            return false;
+    }
+    //INGHRIDY SANTOS  - 05/02/20 - BDEV03 - VALIDAÇÃO SENHA
+    private boolean senhaValida(String senha) {
+        if (senha.length() < 6)
+            return false;
+
+        boolean achouNumero = false;
+        boolean achouMaiuscula = false;
+        boolean achouMinuscula = false;
+        boolean achouSimbolo = false;
+
+        for (char c : senha.toCharArray()) {
+            if (c >= '0' && c <= '9')
+                achouNumero = true;
+            else if (c >= 'A' && c <= 'Z')
+                achouMaiuscula = true;
+            else if (c >= 'a' && c <= 'z')
+                achouMinuscula = true;
+            else
+                achouSimbolo = true;
+        }
+        return achouNumero && achouMaiuscula && achouMinuscula && achouSimbolo;
+    }
+
+    private boolean senhaIguais(String senha, String confirmacaoSenha) {
+        if (confirmacaoSenha.equals(senha))
+            return true;
+        else
+            return false;
+    }
+
+    private boolean ehVazio(String valorComparacao) {
+        if (valorComparacao.isEmpty())
+            return true;
+        else
+            return false;
+    }
+
+
+    //INGHRIDY SANTOS  - 05/02/20 - BDEV02 - NOTIFICAÇÕES
+    protected void notificacao() {
+        Context contexto = getApplicationContext();
+        String textoNotificacao = "Por favor, preencha todos os campos";
+        int duracaoNotifacao = Toast.LENGTH_LONG;
+
+        Toast toast = Toast.makeText(contexto, textoNotificacao, duracaoNotifacao);
+        toast.show();
+    }
+
+    protected void notificacaoNomeCompleto() {
+        Context contexto = getApplicationContext();
+        String textoNotificacao = "Por favor, preencha o nome completo para cadastro no Marvel Time";
+        int duracaoNotifacao = Toast.LENGTH_LONG;
+
+        Toast toast = Toast.makeText(contexto, textoNotificacao, duracaoNotifacao);
+        toast.show();
+    }
+
+    protected void notificacaoSenhaInvalida() {
+        Context contexto = getApplicationContext();
+        String textoNotificacao = "A senha deve conter pelo menos 6 caracteres, uma letra maiuscula e minuscula, um número e um caracter especial.";
+        int duracaoNotifacao = Toast.LENGTH_LONG;
+
+        Toast toast = Toast.makeText(contexto, textoNotificacao, duracaoNotifacao);
+        toast.show();
+    }
+
+    protected void notificacaoSenhaDiferente() {
+        Context contexto = getApplicationContext();
+        String textoNotificacao = "Os dados inseridos nos campos Senha e Confirme sua senha, devem ser iguais. Tente novamente.";
+        int duracaoNotifacao = Toast.LENGTH_LONG;
+
+        Toast toast = Toast.makeText(contexto, textoNotificacao, duracaoNotifacao);
+        toast.show();
+    }
+
+    protected void notificacaoParaProximaTela() {
+        Context contexto = getApplicationContext();
+        String textoNotificacao = "PARABÉNS, em instantes você será direcionado a próxima tela.";
```

Comment:
Uma sugestão para essa notificação pensando na experiência do usuário, seria tirar as palavras em instantes, pois pode passar para o usuário a impressão de que esta ação irá demorar.

Reply:
Na verdade essa notificação de próxima tela deve ser tirada por quem for fazer a transição entre as telas.
Como no meu codigo não contempla a próxima tela, então para testar se o botão está funcionando e deixar um "alerta" de que isso deve ser mudado, fiz uma notificação.

Reply:
Ok, entendido!